exchange/utils.go

```diff
@@ -82,8 +83,29 @@ func (rs *requestSplitter) cleanOpenRTBRequests(ctx context.Context,
 		return
 	}
 
+	gdprSignal, err := getGDPR(req)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	channelEnabled := auctionReq.TCF2Config.ChannelEnabled(channelTypeMap[auctionReq.LegacyLabels.RType])
+	gdprEnforced := enforceGDPR(gdprSignal, gdprDefaultValue, channelEnabled)
+	dsaWriter := dsa.Writer{
+		Config:      auctionReq.Account.Privacy.DSA,
+		GDPRInScope: gdprEnforced,
+	}
+	if err := dsaWriter.Write(req); err != nil {
+		errs = append(errs, err)
```

Comment:
Should we return if there is an error? Same for RebuildRequest and getAuctionBidderRequests.

Reply:
I think we should return an error for `RebuildRequest` but not for the others. `getGDPR` will only ever result in a wrapper unmarshal error and it should theoretically never happen here since any problem should be caught upstream. And also we plan on it not returning an error in the near future anyway. `getAuctionBidderRequests` returns some errors that should not result in an early return and others that are accompanied by a nil `allBidderRequests` which prevents the logic that follows from having any effect since it involves looping over `allBidderRequests`.